apps/openssl.c

```diff
@@ -297,6 +297,20 @@ static void list_md_fn(const EVP_MD *m,
     }
 }
 
+static void list_mac_fn(const EVP_MAC *m,
+                        const char *from, const char *to, void *arg)
+{
+    if (m != NULL) {
+        BIO_printf(arg, "%s\n", EVP_MAC_name(m));
+    } else {
+        if (from == NULL)
+            from = "<undefined>";
+        if (to == NULL)
+            to = "<undefined>";
+        BIO_printf((BIO *)arg, "%s => %s\n", from, to);
```

Comment:
The cast isn't necessary, please remove.

Reply:
No problem, I'll remove it in the morning.   This was a copy/paste from other similar routines.  If they need changing too, I'll submit a separate PR.

I managed to get most of GMAC done today, it needs some updates to allow the cipher to be changed more generally but I hope to add it to the new EVP_MAC interface tomorrow....

Reply:
Changed.